cmd/common-methods.go

```diff
@@ -115,6 +117,27 @@ func getSourceStream(alias string, urlStr string, fetchStat bool, sseKey string)
 				metadata[k] = v
 			}
 		}
+		// If our reader is a seeker try to detect content-type further.
+		if s, ok := reader.(io.ReadSeeker); ok {
+			if ctype := metadata["Content-Type"]; ctype == "application/octet-stream" {
```

Comment:
Why do we check and set the file type **only** when the content type metadata is set to the default `"application/octet-stream"`?

That is, we can be paranoid and don't trust the set file type even when it is set to something other than `"application/octet-stream"` and always set it to the detected file type regardless of what content type the source file comes with?

Reply:
because "guessURLContentType" always gives `application/octet-stream` for unrecognized files @ebozduman